src/app/loadout/loadout-utils.ts

```diff
@@ -80,7 +80,7 @@ export function getArmorStats(
   defs: D1ManifestDefinitions | D2ManifestDefinitions,
   items: DimItem[]
 ): { [hash: number]: DimCharacterStat } {
-  const statDefs = armorStats.map((hash) => defs.Stat.get(hash));
+  const statDefs = armorStats.map((hash) => defs.Stat.get(hash) as DestinyStatDefinition);
```

Comment:
I'm confused, why is the cast needed. Shouldn't get be returning type `DestinyStatDefinition`? 

Also random thought, should `defs.Blah.get(hash)` return `Blah | undefined` to be a little safer?

Reply:
1) it gets a d1 stat | a d2 stat, but d1 stat is currently untyped, so it's cast here. i hate it too but i'm not gonna type d1 manifest stat table

2) it makes for a LOT more lines of code. we rely on the manifest being mostly valid lookups. this is why i set up defs lookup failure logging.

Reply:
Cool, that all makes sense. I've wondered about the get's not having the undefined return type before. Now that we have optional chaining would it a more minor change?

Reply:
there's a bunch of cases like
```
b = get(a);
c = get(b.something.hash)
d = get(c.somethingelse.hash)
```

that become super duper full of repetition and mixed types and `&&`s if you chain the possible failures all the way down

Reply:
Yeah ok that is annoying. I'm still inclined to like the optional return for safety but I'm not going to go on a mission to change it.

Reply:
I've taken a whack at it a few times and got overwhelmed, but if you can make it happen that'd be awesome.